### PR TITLE
[feat] Support end-to-end encryption in C Reader API

### LINKED ISSUE #166: [feature request] C Reader api support End-to-end Encryption
## Motivation

Need to expose the `CryptoKeyReader` of C Reader.

### PR DESCRIPTION
**Pull request by rbarbey**
Fixes #166 and is a prerequisite for apache/pulsar-client-node#276

### Motivation

This change allows C and derived clients to use end-to-end encryption for Readers.

### Modifications

Exposed the methods that already in the C++ API in the C interface

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by shibd** [comment]
Nice work! Can you fix the code format and add a unit test?

**Comment by rbarbey** [comment]
@shibd Thank you for the feedback! I've fixed the formatting and wrote a very basic test for the reader config C API, much like the one for the consumer configuration.

If this PR gets approved, is there any chance to still get this into the 3.2.0 release (saw the voting thread on the dev mailing list)? This would be really great since my ultimate goal is address apache/pulsar-client-node#276

File: tests/c/c_ReaderConfigurationTest.cc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <gtest/gtest.h>
+#include <pulsar/c/reader_configuration.h>
+
+#include <climits>
+
+/*
+  ASSERT_EQ(consumerConf.getConsumerType(), ConsumerExclusive);
```
Comment (shibd) [inline tests/c/c_ReaderConfigurationTest.cc:25]: What is this for?

Comment (rbarbey) [reply tests/c/c_ReaderConfigurationTest.cc:25]: 😬 Sorry, this was used for inspiration. I've removed it

**Review by shibd — APPROVED** [review]

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**